File: Cinema Web App/DatabaseHelper.java
package webapp;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.Statement;

public class DatabaseHelper {
    private static final String URL = "jdbc:mysql://localhost:3306/cine";
    private static final String USER = "root";
    private static final String PASSWORD = "";

    public Connection connect() {
        Connection connection = null;
        try {
            Class.forName("com.mysql.cj.jdbc.Driver");
            connection = DriverManager.getConnection(URL, USER, PASSWORD);
        } catch (Exception e) {
            e.printStackTrace();
        }
        return connection;
    }

    public void closeConnection(Connection connection) {
        if (connection != null) {
            try {
                connection.close();
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
    }

    public ResultSet executeQuery(String query) {
        Connection connection = connect();
        ResultSet resultSet = null;
        try {
            Statement statement = connection.createStatement();
            resultSet = statement.executeQuery(query);
        } catch (Exception e) {
            e.printStackTrace();
        }
        return resultSet;
    }

    public int executeUpdate(String query, Object... params) {
        Connection connection = connect();
        int result = 0;
        try {
            PreparedStatement preparedStatement = connection.prepareStatement(query);
            for (int i = 0; i < params.length; i++) {
                preparedStatement.setObject(i + 1, params[i]);
            }
            result = preparedStatement.executeUpdate();
        } catch (Exception e) {
            e.printStackTrace();
        }
        return result;
    }
}

File: Cinema Web App/DatabaseHelper.php
<?php
class DatabaseHelper {
    private $servername = "localhost";
    private $username = "root";
    private $password = "";
    private $dbname = "cine_DB";
    public $conn;

    public function __construct() {
        $this->conn = new mysqli($this->servername, $this->username, $this->password, $this->dbname);
        if ($this->conn->connect_error) {
            die("Conexión fallida: " . $this->conn->connect_error);
        }
    }

    public function executeQuery($query) {
        return $this->conn->query($query);
    }

    public function executeUpdate($query, $params) {
        $stmt = $this->conn->prepare($query);
        $stmt->bind_param("iiii", ...$params);
        return $stmt->execute();
    }

    public function __destruct() {
        $this->conn->close();
    }
}
?>

